src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/LiteHash.Windows.cs

```diff
@@ -221,36 +245,33 @@ static Exception FailThrow(string algorithm)
             }
         }
 
-        [MemberNotNull(nameof(_hashHandle))]
-        public void Reset()
-        {
-            _hashHandle?.Dispose();
-            SafeBCryptHashHandle hashHandle;
-
-            NTSTATUS ntStatus = Interop.BCrypt.BCryptCreateHash(
-                _algorithm,
-                out hashHandle,
-                pbHashObject: IntPtr.Zero,
-                cbHashObject: 0,
-                secret: ReadOnlySpan<byte>.Empty,
-                cbSecret: 0,
-                BCryptCreateHashFlags.None);
+        public void Reset() => Finalize(Span<byte>.Empty);
```

Comment:
It's cheaper to implement `Reset` in terms of `Final` with an empty destination instead of destroying and re-creating the handle. Bonus points that it makes all of the structs `readonly` now.